### DISCUSSION TITLE
How do I pass additional arguments to callback functions?

### DISCUSSION THREAD

**Discussion by AleNovelli** [python-telegram-bot/python-telegram-bot#3880] [Q&A] [answered]
I am tring to create a class that will manage my telegram bot and will handle interaction with other classes.
However I can't find a way to pass `self` to the functions called in CommandHandler.

Here is a minimal example of the code I would like to be able to run; however it is not possible to pass the `args` argument to `CommandHandler`.
```
# minimal bot class

from telegram.ext import Application, CommandHandler

class Minimal_Example_Bot():
    def __init__(
            self,
            token,
            name:str,
            ):
        self.token=token
        self.name=name
        self.people_greeted=0

    async def hi(self,update,context):
        self.people_greeted+=1
        await update.message.reply_text("{} says: 'Hi!!!'  ".format(self.name))
       
    def start(self):
        application = Application.builder().token(self.token).build()
        application.add_handler(CommandHandler("hi", self.hi, args=[self]))
        application.run_polling()
       
bot=Minimal_Example_Bot(token="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx", name="BestBot")
bot.start()
```
Expected output: `BestBot says: 'Hi!!!`

Do you know how I could achieve such a behavior? I would like to not have to add a bunch of `global` variables and functions.

**Comment by Bibo-Joshi** [comment, marked as answer]
There is no need to pass anything extra. `self.hi` is a bound method of `self`, so you can readily use `self` in the method.

**Reply by AleNovelli** [reply]
Thanks, the solution was easier than I thought
